src/briefcase/commands/base.py

```diff
@@ -876,16 +871,39 @@ def update_cookiecutter_cache(self, template: str, branch="master"):
         if is_repo_url(template):
             # The app template is a repository URL.
             #
-            # When in `no_input=True` mode, cookiecutter deletes and reclones
-            # a template directory, rather than updating the existing repo.
+            # When in `no_input=True` mode, cookiecutter deletes and reclones a template
+            # directory, rather than updating the existing repo.
             #
-            # Look for a cookiecutter cache of the template; if one exists,
-            # try to update it using git. If no cache exists, or if the cache
-            # directory isn't a git directory, or git fails for some reason,
-            # fall back to using the specified template directly.
-            cached_template = cookiecutter_cache_path(template)
+            # Look for a Briefcase cache of the template.
+            cached_template = self.template_cache_path(template)
+
+            if cached_template.exists():
+                # There is a pre-existing cache of the template. Attempt to update it;
+                # if the fetch fails, use the existing state of the cache. Any other
+                # failure is surfaced to the user.
+                try:
+                    repo = self.tools.git.Repo(cached_template)
+                except self.tools.git.exc.GitError as e:
+                    raise BriefcaseCommandError(
+                        f"Template repository is in a weird state. Delete {cached_template} and retry."
+                    ) from e
```

Comment:
I think that if Briefcase's own cache for a template is corrupted, it should just delete it for the user instead asking the user to delete a directory deep in to hidden file system structures.

Reply:
That's a good idea - I'll add it in.

Reply:
Actually - on second thought; I'm not sure deleting is a good idea. It might make sense to delete if we could "delete and retry"; but that's not going to be simple to do here (possible, but not trivial). In the case that we *do* have a corrupt repo, it might be helpful to investigate *why* it is corrupt, which we won't be able to do if it's deleted as soon as it's found to be corrupt.

Reply:
That's a good point. In my experience, Git is reasonably resilient once the initial clone is complete, so we only need to deal with cleanup in the the "new repo" case. 

I'll add `KeyboardInterrupt` protection around the initial new repo clone, as well ensuring a clone failure on the *initial* failure does cleanup; if there's a failure past that point, there's definitely something wonky going on, and we probably want to investigate before cleaning up.

Reply:
> > Actually - on second thought; I'm not sure deleting is a good idea. It might make sense to delete if we could "delete and retry"; but that's not going to be simple to do here (possible, but not trivial).
> 
> I'm not sure I'm following....can't we just fall in to "first time clone" logic below if `GItPython` falls to detect the template as a repo and we've deleted whatever might be there? Is there something I'm missing?

Huh - I hadn't considered that - if I turn the "if/else" into an "if exists; if not exists", the re-clone can follow the failed update. It's obvious now I look at it. I'll add that change/